railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -164,9 +164,12 @@ def config_when_updating
         end
       end
 
-      if !skip_sprockets?
-        insert_into_file "config/application.rb", %(require "sprockets/railtie"), after: /require\(["']rails\/all["']\)\n/
+      unless defined?(Sprockets::Rails)
+        @generator.gem "sprockets-rails"
```

Comment:
Does this add sprockets-rails to gemfile?

Reply:
Yep! https://api.rubyonrails.org/classes/Rails/Generators/Actions.html#method-i-gem